frontend/src/httpLayers/registration.http.js

```diff
@@ -0,0 +1,14 @@
+import wof_api from "./wofapi";
+
+async function registerUser(username, password, email) {
+  await wof_api.post(
```

Comment:
I think that you should do 'try {} catch()' error handling and this function should return code '201' on success and code '400' with error message on failure.

Reply:
Did it a bit differently after all. I have a strong feeling that we shouldn't leak any status codes past this function. The interaction with http requests should end in here and the returned values should be agnostic to the underlying response or lack thereof from the backend service.

Reply:
But why? What is wrong with return status here? It would be easier

Reply:
It leaks the http layer. It won't be easier because it creates a situation where you need a bunch of ifs if you want to ensure the result worked. Moreover, you want to have access to the error message if something goes wrong, which means you need to return an object. It's not easier. Right now, it either returns true or throws. The caller can catch the error and deal with it how they see fit.

Reply:
We want to isolate the contact with the backend service, the logic associated with it from the frontend application. Returning a status is breaking that isolation, which down the line will create a bunch of problems. Imagine having to refactor this function and its returns - you would have to refactor the logic in the frontend app dealing with the returns. Right now, the contract is easier to maintain.